### PR TITLE
Deprecate the quarkus-resteasy-mutiny extension

### PR DESCRIPTION
**Pull request by cescoffier**
Also added a warning explaining why it's deprecated.

### CODE REVIEW

**Review by geoand — APPROVED** [review]

File: extensions/resteasy-classic/resteasy-mutiny/deployment/src/main/java/io/quarkus/resteasy/mutiny/deployment/ResteasyMutinyProcessor.java [hunk 1]
```diff
@@ -8,7 +10,16 @@
...
+        warn();
         return new FeatureBuildItem(Feature.RESTEASY_MUTINY);
     }
 
+    void warn() {
+        Logger.getLogger(ResteasyMutinyProcessor.class).warn("The quarkus-resteasy-mutiny extension is deprecated. " +
+                "Switch to RESTEasy Reactive instead.\n" +
+                "This extension adds support for Uni and Multi to RESTEasy 'classic', without using the reactive execution model,"
+                +
+                " as RESTEasy classic does not use it. To properly integrate Mutiny and RESTEasy, use RESTEasy reactive.");
```
Comment (gastaldi) [inline extensions/resteasy-classic/resteasy-mutiny/deployment/src/main/java/io/quarkus/resteasy/mutiny/deployment/ResteasyMutinyProcessor.java:22]:
```suggestion
                " as RESTEasy classic does not use it. To properly integrate Mutiny and RESTEasy, use RESTEasy Reactive. See https://quarkus.io/guides/getting-started-reactive for more detailed instructions");
```

Not sure if the link is the correct one of if we should have a guide explaining how to migrate

**Comment by gsmet** [comment]
I wonder if we should sneak this one in into 2.7? Maybe worth asking @tqvarnst .

**Review by Sanne — APPROVED** [review]

**Review by gastaldi — APPROVED** [review]

File: extensions/resteasy-classic/resteasy-mutiny/deployment/src/main/java/io/quarkus/resteasy/mutiny/deployment/ResteasyMutinyProcessor.java [hunk 2]
```diff
@@ -8,7 +10,16 @@
 
     @BuildStep
     FeatureBuildItem feature() {
+        warn();
         return new FeatureBuildItem(Feature.RESTEASY_MUTINY);
     }
 
+    void warn() {
+        Logger.getLogger(ResteasyMutinyProcessor.class).warn("The quarkus-resteasy-mutiny extension is deprecated. " +
```
Comment (gastaldi) [inline extensions/resteasy-classic/resteasy-mutiny/deployment/src/main/java/io/quarkus/resteasy/mutiny/deployment/ResteasyMutinyProcessor.java:18]: Maybe it's easier to use `io.quarkus.logging.Log.warn` here, but this is fine

**Comment by geoand** [comment]
Good point, added

**Pull request merged**